sdk/src/azure/core/az_context.c

```diff
@@ -33,6 +35,8 @@ AZ_NODISCARD int64_t az_context_get_expiration(az_context const* context)
 AZ_NODISCARD az_result
```

Comment:
nit: Remove the duplicated comment above, in source. This comment belongs in the header file (and already exists).

It might be helpful to the caller to know that the search is happening by comparing the pointer references, and not by their value. Consider adding that `@remark` to the doc comments and add that to tests.

```C
void const* const key = "k";
void const* const value = "v";
az_context ctx1 = az_context_create_with_expiration(&az_context_application, 100);
az_context ctx2 = az_context_create_with_value(&ctx1, key, value);
az_context ctx3 = az_context_create_with_expiration(&ctx2, 250);

int64_t expiration = az_context_get_expiration(&ctx3);
void const* value2 = NULL;

// This returns AZ_ERROR_ITEM_NOT_FOUND, not AZ_OK, which is the intended behavior.
void const* const a_different_key = "k";
az_result r = az_context_get_value(&ctx3, a_different_key, &value2);

assert_true(r == AZ_OK); // Fails
assert_true(value2 == NULL);
```